### DISCUSSION TITLE
map sizes

### DISCUSSION THREAD

**Discussion by Buhler0014** [KillBait/AutoDrive_Course_Editor#38] [Ideas]
Is there a way to make it so all these all new map size 1x, 4x, 6x, 8x, 16x, can be supported

**Comment by Dzi4d3k** [comment]
6x and 8x? AFAIK the size of map can be only 2^n of normal 2048x2048 (pixels) size.

**Reply by Buhler0014** [reply]
Have a look at PMC_Super_Six_6km I tried all the pre-sets but none are lining up

**Comment by Buhler0014** [comment]
Have a look at PMC_Super_Six_6km I tried all the presets but none are lining up

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

**Comment by Dzi4d3k** [comment]
Wow, indeed, that's great, but actually size map in notation of multiplies is not the same as map size in km.
Look that PMC calls them 6km, 10km, 32km, so for example:

> PMC Super Six 6km
> Terrain size: 6144 meters x 6144 meters, which is 6.1 kilometers x 6.1 kilometers (elmcreek is 2km). Terrain type: american, fictional. 

That would mean that map 6km is actually map 9x standard size, that mean 2kmx2km fits 9 times on map 6kmx6km.
Accordingly map 10km is 25x time larger that standard ElmCreek, and 32km map is INCREDIBLY **256x bigger** than standard map.

Nevertheless all above is excellent examples of creativity. Now I'm not sure how @KillBait will react on this 😄

**Comment by Buhler0014** [comment]
Cool just wanted to bring it up. Not true was trying to edit routes lol

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

**Comment by KillBait** [comment]
I'm not surprised that these map sizes appeared.

In theory the scaling code in the editor should be able to account for the odd sizes, but, it was originally coded with 1,4 and 16x being the only accepted sizes.

I'm currently away from home on a 3 week training course at the moment and I don't have access to a dev environment, so I can't test if my theory is correct, but when I get back I'll look into it.

It may be a small amount of work to support these sizes, or maybe not, but I'll let you know once I'm able to get back home and test it

**Comment by KillBait** [comment]
Well.. i did a bit of paper math since i was bored in my hotel room and i have another week until i'm back home.

If you take the standard 1x/4x/16x naming convention ( 2^n ) that is used for FS map sizes, things get totally
ludicrous with these map sizes. The naming convention goes crazy. 

1x      = 2048 x 2048  = 2km ( 1x in editor ) ( e.g. ElmCreek map )
4x      = 4096 x 4096  = 4km ( 4x in editor )
8x      = 6144 x 6144  = 6km ( [PMC Super Six 6km](https://www.farming-simulator.org/22/pmc-super-six-6km.php) )
16x    = 8096 x 8096  = 8km ( 16x in editor )
32x    = 10240 x 10240 = 10km ( [PMC RedWhiteBlue 10km](https://www.farming-simulator.org/22/pmc-redwhiteblue-10km.php) )
64x    = 12288 x 12288 = 12km
128x  = 14366 x 14366 = 14km
256x  = 16384 x 16384 = 16km
512x  = 18432 x 18432 = 18km
1,024x = 20480 x 20480 = 20km ( [PMC Undefined Farms 20km](https://www.farming-simulator.org/22/pmc-undefined-farms-20km.php) )
2,048x = 22528 x 22528 = 22km 
4,096x = 24576 x 24576 = 24km
8,196x = 26624 x 26624 = 26km
16,392x = 28672 x 28672 = 28km
32,784x = 30720 x 30720 = 30km
65,535x = 32768 x 32768 = 32km ( [PMC Cereal Region 32km](https://www.farming-simulator.org/22/pmc-cereal-region-32km.php) )
131,072x = 34816 x 34816 = 34km
262,144x = 36864 x 36864 = 36km
524,288x = 38912 x 38912 = 38km
1,048,576x = 40960 x 40960 = 40km ( [PMC Korkscrew 40km](https://www.farming-simulator.org/22/pmc-korkscrew-40km.php) )
2,097,152x = 43008 x 43008 = 42 km
4,194,304x = 45056 x 45056 = 44 km
5,931,642x = 46080 x 46080 = 45km ( [PMC King Corn 45km](https://www.farming-simulator.org/22/pmc-king-corn-45km.php) ) ( does not match stated map size of 45056 x 45056 )

So...... by the standard 2^n naming convention the largest size map is a 5,931,642x map !!!!!.

Either i add all these crazy million x numbers to the list of map sizes in the menu, or , do i change the map size menu to
the km designation, but, since the regular maps still use the 1x/4x/16x i'm undecided what to do.

Do i ignore these maps since there such a niche size, or add them regardless..

Any comments?

**Comment by Buhler0014** [comment]
I think that would be way to much work for the amount of maps that are coming at this point or if it is easier to go to a km size set would likely be a better option if you decide to. But I would like to add I am not asking you to do so you are moder it is your time put in I just get to reap the reward of your hard work. I thank you for that and for even looking into how much work it would be.

**Comment by KillBait** [comment]
My compromise would be to change the size menu to a list like this, and it should'nt be a lot of work to add them

- 2km ( 1x )
- 4km ( 4x )
- 6km ( 8x )
- 8km ( 16x )
- 10 km ( 32x )
- ect.....

**Reply by Dzi4d3k** [reply]
@KillBait 
Not sure how you done your math but, for example:

6km map is = 6km wide on 6km height, which is ( 3x 2km) x (3x 2km), what gives 9x map 
accordingly
32km map is = (16x 2km) x (16x 2km), what gives 256x map grater then standard
and the largest one:
45km map is (22,5x 2km) x (22,5x 2km), what is 506,25 grater than standard.
All this is possible by stretching density maps so one pixel cover couple of meters (not remember now from by head how much exactly), bu the point is that playing on such inaccurate terrain gives awkward felling when combine cut crops few meters away from edge of header.

**Comment by Buhler0014** [comment]
If that is your choice, not mine, then great. I do not do that many just ones I am asked specifically for and I always give them your discord for ad routes first or check for one released but will make changes to route for people as a lot don’t have a clue how to do it or to lazy to figure out how.

**Comment by KillBait** [comment]
There are other factors involved in supporting these large map sizes

(1) menu edits - easy to do
(2) the max zoom limit will have to be made configurable,  the current max isn't enough to easily see the space between the nodes - easy to do
(3) standard heightmaps are 1025x1025, but these bigger maps have 4096x4096+ heightmaps - shouldn't be too much work

The downsides are.

(1) memory usage of the editor will go up the higher the map size
(2) the large networks these maps could have will slow down the render speeds

I'm not against adding support for them, it's just they will come at a heavy cost ( memory and speed ) wise. I will add a warning
if larger sizes are used, just to make people aware that it has downsides.

**Comment by Buhler0014** [comment]
I just used what they said was size and checked what they had for pixel size at first had a hard time figuring it out as such weird sizes some still don’t make sense or are not set to proper pixel

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

________________________________

**Comment by Buhler0014** [comment]
Is the cost worth it for the number of maps coming so far I only know of 2 guys or groups making them. Are they going to catch on. For me a 16x is huge enough. I would also think a 32x if a number of people playing would be plenty big enough. For me I wouldn’t want to have to make a route map for anything bigger than a 16x or maybe if I was board a 32x which in turn by the time I was finished would be board. For me it not a big deal as I have a great computer but most others don’t.


Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

**Comment by KillBait** [comment]
Hi Everyone

Sorry for the delay and silence, the next version of the editor is nearly ready, i have added support for map sizes
up to 46km and this was more work than i anticipated.

I have just have one problem left to solve, some larger maps use a scaling value on the heightmap info, while
this shouldn't be a big issue, this scaling value is set inside the giants editor and is not accessible from the
saved maps files,

I'm trying to figure out how to reverse engineer this value, i may have it working, but i need a little more time
to test it, hopefully it wont be too much longer.
